File: 이상욱/BOJ/m3w7/BOJ17609.java
package m3w12;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BOJ17609 {

	public static void main(String[] args) throws Exception {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(br.readLine());
		
		for(int i = 0 ; i < T; i++) {
			int ans = 2;
			String string = br.readLine();
			
			if(palinDrome(string)) {
				ans = 0;
			}else if(palinDrome2(string)) {
				ans = 1;
			}
			
			System.out.println(ans);
		}
		
	}

	private static boolean palinDrome(String string) {
		int start = 0;
		int end = string.length()-1;
		
		while(start <= end) {
			if(string.charAt(start) != string.charAt(end)) {
				return false;
			}
			start++;
			end--;
		}
		return true;
	}
	
	private static boolean palinDrome2(String string) {
		int start = 0;
		int end = string.length()-1;
		
		while(start <= end) {
			if(string.charAt(start) != string.charAt(end)) {
				
				boolean res = palinDrome(string.substring(start+1,end+1));
				boolean res1 = palinDrome(string.substring(start,end));
				
				return res || res1 ;
				
			}
			start++;
			end--;
		}
		
		return true;
	}

	
}

File: 이상욱/BOJ/m3w7/BOJ17609.kt
package m3w12

import java.io.BufferedReader
import java.io.InputStreamReader

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))

    val T = br.readLine().toInt()

    repeat(T){
        var ans = 2
        val string = br.readLine()
        if(palinDrome(string)){
            ans = 0
        }else if(palinDrome2(string)){
            ans = 1
        }
        println(ans)

    }
}


fun palinDrome(string: String) : Boolean {
    var start = 0
    var end = string.length-1

    while(start <= end){
        if(string[start] != string[end]){
            return false
        }
        start++
        end--
    }
    return true
}

fun palinDrome2(string: String): Boolean {
    var start = 0
    var end = string.length-1

    while(start <= end){
        if(string[start] != string[end]){

            val res = palinDrome(string.substring(start+1,end+1))
            val res1 = palinDrome(string.substring(start,end))

            return res || res1

        }
        start++
        end--
    }
    return true
}